### PR TITLE
[Improve](map-type) Add contains_null for map 

### PR DESCRIPTION
**Pull request by amorynan**
# Proposed changes

add contains_null for map 

#16547 

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/MapLiteral.java [hunk 1]
```diff
@@ -47,30 +47,18 @@ public MapLiteral(LiteralExpr... exprs) throws AnalysisException {
...
+                throw new AnalysisException("Invalid key type in Map Only support scalar type");
+            }
+            valueType = Type.getAssignmentCompatibleType(valueType, exprs[idx + 1].getType(), true);
+            if (valueType == Type.INVALID) {
+                throw new AnalysisException("Invalid value type in Map");
             }
-            children.add(expr);
-            ++ idx;
+            children.add(exprs[idx]);
+            children.add(exprs[idx + 1]);
```
Comment (xy720) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/MapLiteral.java:61]: Just a example:

```
for (int idx = 0; idx < exprs.length && idx + 1 < exprs.length; idx += 2) {
    if (MapType.supportKeyType(exprs[idx].getType())) {
        throw new AnalysisException("Invalid key type in Map, not support " + keyType);
    }
    keyType = Type.getAssignmentCompatibleType(keyType, exprs[idx].getType(), true);
    valueType = Type.getAssignmentCompatibleType(valueType, exprs[idx + 1].getType(), true);
}

if (keyType == Type.INVALID) {
    throw new AnalysisException("Invalid key type in Map.");
}

if (valueType == Type.INVALID) {
    throw new AnalysisException("Invalid value type in Map.");
}

for (int idx = 0; idx < exprs.length && idx + 1 < exprs.length; idx += 2) {
    if (exprs[idx].getType().equals(keyType)) {
        children.add(expr[idx]);
    } else {
        children.add(expr[idx].castTo(keyType));
    }
    if (exprs[idx+1].getType().equals(valueType)) {
        children.add(expr[idx+1]);
    } else {
        children.add(expr[idx+1].castTo(valueType));
    }
}
```

File: fe/fe-core/src/main/java/org/apache/doris/analysis/MapLiteral.java [hunk 2]
```diff
@@ -47,30 +47,18 @@ public MapLiteral(LiteralExpr... exprs) throws AnalysisException {
...
-                    valueType = Type.getAssignmentCompatibleType(valueType, expr.getType(), true);
-                }
-                if (valueType == Type.INVALID) {
-                    throw new AnalysisException("Invalid element type in Map");
-                }
+        for (int idx = 0; idx < exprs.length && idx + 1 < exprs.length; idx += 2) {
+            // limit key type to scalar type
+            keyType = Type.getAssignmentCompatibleType(keyType, exprs[idx].getType(), true);
+            if (keyType == Type.INVALID || !keyType.isScalarType()) {
+                throw new AnalysisException("Invalid key type in Map Only support scalar type");
```
[the remark this answers is not in the document]
Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/MapLiteral.java]: for example, date and int are scala types but they are not compatible with each other.
But the error msg is "Invalid key type in Map Only support scalar type".

File: fe/fe-core/src/main/java/org/apache/doris/analysis/MapLiteral.java [hunk 3]
```diff
@@ -47,30 +47,35 @@ public MapLiteral(LiteralExpr... exprs) throws AnalysisException {
...
-                    keyType = Type.getAssignmentCompatibleType(keyType, expr.getType(), true);
-                }
-                if (keyType == Type.INVALID) {
-                    throw new AnalysisException("Invalid element type in Map");
-                }
+        // check types here
+        // 1. limit key type with map-key support
+        // 2. check type can be assigment for cast
+        for (int idx = 0; idx < exprs.length && idx + 1 < exprs.length; idx += 2) {
+            if (type.supportSubType(exprs[idx].getType())) {
```
Comment (xy720) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/MapLiteral.java:54]: type is not initialized.
Why not changed this method to a static method?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/MapLiteral.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/MapLiteral.java]: yeh , I see , 
date and int are scala types but they are not compatible with each other. so here should throw err, but the error msg should be invaild key type for int (assume origin key type is date.)

**Review by xy720 — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run compile

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run buildall

**Review by xy720 — APPROVED** [review]
LGTM

**Pull request merged**